### PR TITLE
[states] Move States to vernac

### PR DESCRIPTION
**Pull request by ejgallego**
We continue to push state layers upwards, in preparation of a
functional vernacular interpretation.

Now we move `States` and `Printmod` which messes with the global
state as to temporarily create envs with modules.

### CODE REVIEW

**Review by maximedenes — COMMENTED** [review]
I thought `Printmod` was at the right place in `printing`. Why does it belong to `vernac`?

Of course, it should not mess with the global environment, isn't this the real problem that should be fixed, rather than moving the file?

**Comment by ejgallego** [comment]
> I thought `Printmod` was at the right place in `printing`. Why does it belong to `vernac`?

It would be great indeed to fix that, but this seems like a very complex task. See the commit I've just added, already we had to abstract to imperative operations in `Printmod` as an `~mod_ops` argument.

I think it is simpler to do like we do here _for now_ , until someone has the energy to refactor `Declaremods`, and we remove one more piece of state from the printers.

File: vernac/vernac.mllib [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Vernacexpr
 Attributes
 Pvernac
+States
+Printmod
 Declaremods
```
Comment (SkySkimmer) [inline vernac/vernac.mllib:6]:
```suggestion
Declaremods
Printmod
```

Comment (ejgallego) [reply vernac/vernac.mllib]: Oups! Thanks.

**Review by SkySkimmer — APPROVED** [review]
This works OK IMO, if we see printing/ as being more about the constrexpr -> pp phase than something more ambitious for all printing purposes.

**Comment by ejgallego** [comment]
Needs assignee.

**Comment by SkySkimmer** [comment]
Maybe we will end up putting the mod_ops back with an added state_protect field but for now this should be fine.

**Pull request merged**